base/strings/util.jl

```diff
@@ -438,8 +440,11 @@ end
 Search for the given pattern `pat` in `s`, and replace each occurrence with `r`.
 If `count` is provided, replace at most `count` occurrences.
 `pat` may be a single character, a vector or a set of characters, a string,
-or a regular expression. If `r`
+or a regular expression.
+If `pat` is a `Regex` or `AbstractString` and `r`
 is a function, each occurrence is replaced with `r(s)` where `s` is the matched substring.
+If `pat` is a `Char`, a collection of `Char` or a predicate function and `r`
```

Comment:
Instead of repeating the full sentence, maybe just change ``where `s` is the matched substring.`` to ``where `x` is the matched substring (when `pat` is a `Regex` or `AbstractString`) or character (when `pat` is a `Char` or a collection of `Char`).``?